collectors/base.py

```diff
@@ -23,15 +40,63 @@ def _admin_socket(self, cmds=None, socket_path=None):
             cmds = ['perf', 'dump']
 
         start = time.time()
-        response = admin_socket(adm_socket, cmds,
-                                format='json')
+        try:
+            response = admin_socket(adm_socket, cmds,
+                                    format='json')
+        except RuntimeError as e:
+            self.logger.error("admin_socket error: {}".format(e.message))
+            self.error = True
+            self.error_msgs = [e.message]
+            resp = {}
+        else:
+            resp = json.loads(response)
+
         end = time.time()
 
         self.logger.debug("admin_socket call '{}' : "
                           "{:.3f}s".format(' '.join(cmds),
                                            (end - start)))
 
-        return json.loads(response)
+        return resp
+
+    def get_version(self):
+        """
+        Although the version number is v.r.m based, this isn't a float so it
+        can't be stored as a number, so the version returned is just the
+        vesion.release components (i.e. looks like a float!)
+
+        :return: version number (float)
+        """
+        # version command returns output like this
+        # ceph version 10.2.2-15.el7cp (60cd52496ca02bdde9c2f4191e617f75166d87b6)
+
+        cmd = BaseCollector.class_to_cmd.get(self._name, 'ceph')
+        vers_command = "{} --version".format(cmd)
+        vers_output = os_cmd(vers_command)
+        if vers_output:
+            return float('.'.join(vers_output.split()[2].split('.')[:2]))
```

Comment:
it needs to be a float to pass into graphite - text is not acceptable, which leaves int or float.